dask/dataframe/csv.py

```diff
@@ -0,0 +1,174 @@
+
+    This accepts a list of lists of values of bytes where each list corresponds
+    to one file, and the value of bytes concatenate to comprise the entire
+    file, in order.
+
+    Parameters
+    ----------
+    block_lists: list of lists of delayed values of bytes
+        The lists of bytestrings where each list corresponds to one logical file
+    header: bytestring
+        The header, found at the front of the first file, to be prepended to
+        all blocks
+    head: pd.DataFrame
+        An example Pandas DataFrame to be used for metadata.
+        Can be ``None`` if ``collection==False``
+    kwargs: dict
+        Keyword arguments to pass down to ``pd.read_csv``
+    collection: boolean, optional (defaults to True)
+
+    Returns
+    -------
+    A dask.dataframe or list of delayed values
+    """
+    dtypes = head.dtypes.to_dict()
+    columns = list(head.columns)
+    dfs1 = [[delayed(bytes_read_csv)(b, header, kwargs, dtypes, columns)
+              for b in blocks]
+              for blocks in block_lists]
+    dfs2 = sum(dfs1, [])
+
+    if collection:
+        return from_delayed(dfs2, head)
+    else:
+        return dfs2
+
+
+def read_csv(filename, blocksize=2**25, chunkbytes=None,
+        collection=True, lineterminator='\n', compression=None,
+        enforce_dtypes=True, sample=10000, **kwargs):
+    """ Read CSV files into a Dask.DataFrame
+
+    This parallelizes the ``pandas.read_csv`` file in the following ways:
+
+    1.  It supports loading many files at once using globstrings as follows:
+
+        >>> df = dd.read_csv('myfiles.*.csv')  # doctest: +SKIP
+
+    2.  In some cases it can break up large files as follows:
+
+        >>> df = dd.read_csv('largefile.csv', blocksize=25e6)  # 25MB chunks  # doctest: +SKIP
+
+    Internally dd.read_csv uses pandas.read_csv and so supports many of the
+    same keyword arguments with the same performance guarantees.
+
+    See the docstring for ``pandas.read_csv`` for more information on available
+    keyword arguments.
+
+    Parameters
+    ----------
+
+    filename: string
+        Filename or globstring for CSV files.  May include protocols like s3://
+    blocksize: int or None
+        Number of bytes by which to cut up larger files
+    collection: boolean
+        Return a dask.dataframe if True or list of dask.delayed objects if False
+    sample: int
+        Number of bytes to use when determining dtypes
+    **kwargs: dict
+        Options to pass down to ``pandas.read_csv``
+    """
+    if chunkbytes is not None:
+        warn("Deprecation warning: chunksize csv keyword renamed to blocksize")
+        blocksize=chunkbytes
+    if 'index' in kwargs or 'index_col' in kwargs:
+        raise ValueError("Keyword 'index' not supported "
+                         "dd.read_csv(...).set_index('my-index') instead")
+
+    if blocksize and compression not in seekable_files:
+        print("Warning %s compression does not support breaking apart files\n"
```

Comment:
Using `print` for warnings bothers me a bit as it can't be turned off. It also goes to `stdout` instead of `stderr`, so will be hidden if the user is piping `stdout` to file. Can we switch to `warn` here?